packages/ui-react/src/containers/ShelfList/ShelfList.tsx

```diff
@@ -1,88 +1,136 @@
+import { isScheduledOrLiveMedia } from '@jwp/ott-common/src/utils/liveEvent';
 
 import ShelfComponent from '../../components/Shelf/Shelf';
 import InfiniteScrollLoader from '../../components/InfiniteScrollLoader/InfiniteScrollLoader';
-import PlaylistContainer from '../PlaylistContainer/PlaylistContainer';
+import ErrorPage from '../../components/ErrorPage/ErrorPage';
 
 import styles from './ShelfList.module.scss';
 
-const INITIAL_ROW_COUNT = 6;
+const INITIAL_ROWS_LOADED = 6;
 const LOAD_ROWS_COUNT = 4;
 
 type Props = {
   rows: Content[];
 };
 
+const placeholderData = generatePlaylistPlaceholder(30);
+
 const ShelfList = ({ rows }: Props) => {
   const { accessModel } = useConfigStore(({ accessModel }) => ({ accessModel }), shallow);
-  const [rowCount, setRowCount] = useState(INITIAL_ROW_COUNT);
+  const [rowsLoaded, setRowsLoaded] = useState(INITIAL_ROWS_LOADED);
+  const { t } = useTranslation('error');
 
   const watchHistoryDictionary = useWatchHistoryStore((state) => state.getDictionaryWithSeries());
 
   // User
   const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
 
+  // Todo: move to more common package?
+  const page_limit = PLAYLIST_LIMIT.toString();
+  const queryClient = useQueryClient();
+  const apiService = getModule(ApiService);
+  const playlists = useQueries(
+    rows.map(({ contentId, type }, index) => ({
+      enabled: !!contentId && rowsLoaded > index,
+      queryKey: ['playlist', contentId],
+      queryFn: async () => {
+        if (type === PersonalShelf.Favorites) return FavoritesStore.getState().getPlaylist();
+        if (type === PersonalShelf.ContinueWatching) return WatchHistoryStore.getState().getPlaylist();
+
+        const playlist = await apiService.getPlaylistById(contentId, { page_limit });
+
+        // This pre-caches all playlist items and makes navigating a lot faster.
+        playlist?.playlist?.forEach((playlistItem) => {
+          queryClient.setQueryData(['media', playlistItem.mediaid], playlistItem);
+        });
+
+        return playlist;
+      },
+      placeholderData: !!contentId && placeholderData,
+      refetchInterval: (data: Playlist | undefined) => {
+        if (!data) return false;
+
+        const autoRefetch = isTruthyCustomParamValue(data.refetch) || data.playlist.some(isScheduledOrLiveMedia);
+
+        return autoRefetch ? 1000 * 30 : false;
+      },
+      retry: false,
+    })),
+  );
+
   useEffect(() => {
     // reset row count when the page changes
-    return () => setRowCount(INITIAL_ROW_COUNT);
+    return () => setRowsLoaded(INITIAL_ROWS_LOADED);
   }, [rows]);
 
+  // If all playlists are empty, due to geo restrictions, we show a geo block error page
+  const allPlaylistsEmpty = playlists.every(({ data, isSuccess }) => isSuccess && !data?.playlist?.length);
+
+  if (allPlaylistsEmpty) {
+    return <ErrorPage title={t('geo_blocked_heading')} message={t('geo_blocked_description')} />;
```

Comment:
Minor 🐜 ; we assume that geo-blocking is the reason for the empty shelves. But it could be a misconfiguration as well. The translation is correct, but the key indicates that it is a geo-block error. 

```suggestion
    return <ErrorPage title={t('empty_shelves_heading')} message={t('empty_shelves_description')} />;
```

Reply:
Good point. We do say "... in your region" in the translations. Are we sure this is sufficient then?